Source/Blazorise/Components/Validation/Validations.razor.cs

```diff
@@ -102,23 +102,41 @@ private async Task<bool> TryValidateAll()
         return validated;
     }
 
-    internal void NotifyValidationInitialized( IValidation validation )
+    /// <summary>
+    /// Notifies the validation system that a new validation component has been initialized and adds it to the list of validations if not already present.
+    /// </summary>
+    /// <param name="validation">The validation component to add.</param>
+    public void NotifyValidationInitialized( IValidation validation )
     {
         if ( !validations.Contains( validation ) )
         {
             validations.Add( validation );
         }
     }
 
-    internal void NotifyValidationRemoved( IValidation validation )
+    /// <summary>
+    /// Notifies the validation system that a validation component is being removed and removes it from the list of validations if present.
+    /// </summary>
+    /// <param name="validation">The validation component to remove.</param>
+    public void NotifyValidationRemoved( IValidation validation )
     {
         if ( validations.Contains( validation ) )
         {
             validations.Remove( validation );
         }
     }
 
-    internal void NotifyValidationStatusChanged( IValidation validation )
+    /// <summary>
+    /// Notifies the validation system that the status of a validation component has changed. This method handles the logic for updating the overall validation status based on the mode (Auto or Manual) and the current validation results.
+    /// </summary>
+    /// <param name="validation">The validation component whose status has changed.</param>
+    /// <remarks>
+    /// In Auto mode, this method triggers the aggregation of validation results and potentially raises a status changed event.
+    /// It is designed to minimize the number of status changed events by aggregating validation results.
+    /// Special consideration is needed to ensure that the status changed event is raised only once per validation cycle,
+    /// even if multiple validations fail.
+    /// </remarks>
+    public void NotifyValidationStatusChanged( IValidation validation )
```

Comment:
Is there a reason why this takes in an `IValidation`? It doesn't seem to be used?

Reply:
I was used at some point but it got not-used over time. I would leave it just in case we need it and that we don't brake the API if we do need it.

Reply:
The problem is that it is now a public api... again, the consumer might be looking at this puzzled trying to pass in the validation and he does not know it's not required at all.

Does this add unnecessary extra effort for the consumer?

Reply:
I've propagated the argument to be used as an extra information for the ValidationsStatusChangedEventArgs. In case anyone needs it, and we don't need to remove it now. Agree?